src/modules/lua_general.c

```diff
@@ -878,8 +878,8 @@ lua_repl_prompt(EditLine *el) {
 
   if(!pthread_equal(eventer_get_owner(ncct->e), pthread_self()))
     snprintf(info->prompt, sizeof(info->prompt), "lua_general(...)# ");
-  else
-    snprintf(info->prompt, sizeof(info->prompt), tl, get_eventer_id(ncct), lmc->lua_state);
+  else 
+    if(lmc)  snprintf(info->prompt, sizeof(info->prompt), tl, get_eventer_id(ncct), lmc->lua_state);
```

Comment:
This would allow us returning without filling out info->prompt.

This shold be moved up to the first if:

``` if(!lmc || !pthread_equal(..... ```

and the else remain as it was.